packages/web/src/common/store/cache/tracks/utils/reformat.ts

```diff
@@ -61,6 +66,31 @@ const setDefaultFolloweeSaves = <T extends TrackMetadata>(track: T) => {
   }
 }
 
+const setTypedPremiumConditions = <T extends TrackMetadata>(track: T) => {
+  const premium_conditions = track.premium_conditions as any
```

Comment:
Yeah, I think since we're using the same type for the metadata in both the response and the reformatted version, we probably can't get around typecasting here.

This might be a place where we are supposed to use `unknown` instead of `any`, to indicate that the type isn't safe and we need to check it before using.

But otherwise, our only recourse would be to change the type used at the api layer to be different from `TrackMetadata` (I believe that would be in `retrieveTracks`). That seems _wayyyy_ beyond scope of what we're doing here and probably requires more planning.  So I think I'm okay with this for now.

cc @dylanjeffers @sliptype (We should consider if we want to separate the types for api responses vs. post-processing types)